### PR TITLE
Add miden user docs

### PR DESCRIPTION
**Pull request by tnachen**
Merged the miden assembly hackmd docs into our user docs, also create the initial structure in docs folder

### CODE REVIEW

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -3,5 +3,15 @@
 [Miden](./README.md)
 - [Introduction](./intro.md)
 - [User Documentation](./user_docs.md)
+  - [Miden VM Overview](./user_docs/overview/vm_overview.md)
+  - [Miden Assembly](./user_docs/assembly/miden_assembly.md)
```
Comment (bobbinth) [inline docs/src/SUMMARY.md:7]: For the root files - should we use some standard naming convention? e.g. maybe `vm_overview.md` should be `main.md`? And same for `miden_assembly.md`?

File: docs/src/SUMMARY.md [hunk 2]
```diff
@@ -3,5 +3,15 @@
...
+    - [Code Organization](./user_docs/assembly/code_organization.md)
+    - [Flow Control](./user_docs/assembly/flow_control.md)
+    - [Field Operations](./user_docs/assembly/field_operations.md)
+    - [u32 Operations](./user_docs/assembly/u32_operations.md)    
+    - [Input / Output Operations](./user_docs/assembly/io_operations.md)
+    - [Cryptographic Operations](./user_docs/assembly/cryptographic_operations.md)
+    - [Debugging](./user_docs/assembly/debugging.md)    
+  - [Miden Standard Library](./miden_stdlib.md)
```
Comment (bobbinth) [inline docs/src/SUMMARY.md:15]: Should we have the same general structure as for Miden assembly? E.g., "./user_docs/stdlib/main.md`?

File: docs/src/miden_assembly.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Miden Assembly
```
Comment (bobbinth) [inline docs/src/miden_assembly.md:1]: This is file needed here?

File: docs/src/SUMMARY.md [hunk 4]
```diff
@@ -3,5 +3,15 @@
 [Miden](./README.md)
 - [Introduction](./intro.md)
 - [User Documentation](./user_docs.md)
```
Comment (bobbinth) [inline docs/src/SUMMARY.md:5]: Should change the structure as: `./user_docs.md` -> `./user_docs/main.md`?

File: docs/src/user_docs.md [hunk 5]
```diff
@@ -1,3 +1,5 @@
 # User Documentation
 
-TODO
+
+- [Miden Assembly](./miden_assembly.md)
+- [Miden Standard Library](./miden_stdlib.md)
```
Comment (bobbinth) [inline docs/src/user_docs.md:5]: I think these links will need to be updated once we update the structure as described in other comments. At the very least, the Miden Assembly link seems to be broken now.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! Left a few inline comments.

File: docs/src/SUMMARY.md (hunk 1 above)
Comment (tnachen) [reply docs/src/SUMMARY.md:7]: fixed

File: docs/src/SUMMARY.md (hunk 2 above)
Comment (tnachen) [reply docs/src/SUMMARY.md:15]: fixed

File: docs/src/miden_assembly.md (hunk 3 above)
Comment (tnachen) [reply docs/src/miden_assembly.md:1]: No it's auto generated, I can remove it

File: docs/src/SUMMARY.md (hunk 4 above)
Comment (tnachen) [reply docs/src/SUMMARY.md:5]: fixed

File: docs/src/user_docs.md (hunk 5 above)
Comment (tnachen) [reply docs/src/user_docs.md]: yes fixed

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**